src/main/java/uk/gov/laa/ccms/caab/bean/validators/client/FindAddressValidator.java

```diff
@@ -33,25 +35,43 @@ public boolean supports(Class<?> clazz) {
    * @param errors The Errors object to store validation errors.
    */
   @Override
-  public void validate(Object target, Errors errors) {
-    ClientFormDataAddressDetails addressDetails = (ClientFormDataAddressDetails) target;
+  public void validate(final Object target, final Errors errors) {
 
-    if (addressDetails.getNoFixedAbode()) {
-      validateNoFixedAbode(addressDetails, errors);
-    }
+    String country = "";
+    String postcode = "";
+    String houseNameNumber = "";
+
+    if (target instanceof ClientFormDataAddressDetails) {
+      final ClientFormDataAddressDetails addressDetails = (ClientFormDataAddressDetails) target;
+
+      if (addressDetails.getNoFixedAbode()) {
+        validateNoFixedAbode(addressDetails, errors);
+      }
 
-    validateRequiredField("country", addressDetails.getCountry(),
+      country = addressDetails.getCountry();
+      postcode = addressDetails.getPostcode();
+      houseNameNumber = addressDetails.getHouseNameNumber();
+
+    } else if (target instanceof AddressFormData) {
+      final AddressFormData addressDetails = (AddressFormData) target;
+
+      country = addressDetails.getCountry();
+      postcode = addressDetails.getPostcode();
+      houseNameNumber = addressDetails.getHouseNameNumber();
+    }
+    validateRequiredField("country", country,
         "Country", errors);
-    if (StringUtils.hasText(addressDetails.getCountry())) {
-      if (!addressDetails.getCountry().equals("GBR")) {
+    if (StringUtils.hasText(country)) {
```

Comment:
`if (COUNTRY_CODE_GBR.equals(country))`

(then there's no need to do nested if)

Reply:
sorted